target/linux/mediatek/image/filogic.mk

```diff
@@ -391,6 +391,31 @@ define Device/glinet_gl-mt6000
 endef
 TARGET_DEVICES += glinet_gl-mt6000
 
+define Device/glinet_gl-x3000-xe3000-common
+  DEVICE_VENDOR := GL.iNet
+  DEVICE_DTS_DIR := ../dts
+  DEVICE_PACKAGES := kmod-mt7981-firmware mt7981-wo-firmware mkf2fs \
+    kmod-fs-f2fs kmod-hwmon-pwmfan kmod-usb3 kmod-usb-serial-option \
+    kmod-usb-storage kmod-usbip-client kmod-usb-net-qmi-wwan uqmi
```

Comment:
Why do you need `kmod-usbip-client`?

Reply:
This is for the USB tethering:
```
[    5.823286] vhci_hcd vhci_hcd.0: USB/IP Virtual Host Controller
[    5.829246] vhci_hcd vhci_hcd.0: new USB bus registered, assigned bus number 3
[    5.836505] vhci_hcd: created sysfs vhci_hcd.0
[    5.841389] hub 3-0:1.0: USB hub found
[    5.850412] hub 3-0:1.0: 8 ports detected
[    5.855150] vhci_hcd vhci_hcd.0: USB/IP Virtual Host Controller
[    5.861078] vhci_hcd vhci_hcd.0: new USB bus registered, assigned bus number 4
[    5.868554] usb usb4: We don't know the algorithms for LPM for this host, disabling LPM.
[    5.877202] hub 4-0:1.0: USB hub found
[    5.882761] hub 4-0:1.0: 8 ports detected
```

Reply:
USB-IP is a protocol to transport USB bus data over TCP/IP, typically used inside LANs with low and constant latency. It's a nice toy to remote access USB printers, scanners and the like for which no Linux drivers exist. Of course, it can also be useful as a development tool. But I fail to understand how it would be a requirement for USB tethering. Or is the modem offering USB-over-IP-over-PCIe when connected via PCIe? (only? is that even possible? I thought USB 2 is always connected...)

Reply:
I may very well have misunderstood the use of `kmod-usbip-client`. I just tried to replicate the USB devices detected by the GL.iNet firmware, and I noticed that without this module, the USB/IP Virtual Host Controllers I pointed out in the logs were not detected. I'll double-check tomorrow on the device, as I don't have it at hand today.

Reply:
I confirm that removing `kmod-usbip-client` removes the detection of these two USB/IP Virtual Host Controllers, but as I'm not sure what they are used for, I prefer dropping this module.